### ISSUE TITLE
When building with python3, Rodan-main container builds but faces some bugs 

### ISSUE THREAD

**Issue by sabrina0822** [DDMAL/Rodan#708]
Creating this issue to document all the bugs we come across as we attempt to migrate rodan from python 2 to python 3.
Brief update: As of right now, we have been able to migrate Rodan to python 3 and build all containers successfully. Once these images are pulled multiple bugs have been surfacing.

Bug 1: Since manage.py is now in python 3, when running the rodan-main container this pops up.
<img width="777" alt="image" src="https://user-images.githubusercontent.com/90227881/178524804-b0f84411-5a10-4f7e-a4c1-2870aea665d6.png">
Running manage.py makemigrations works without problems when I try to run manage.py migrate this is the traceback that pops up: 
<img width="884" alt="image" src="https://user-images.githubusercontent.com/90227881/178526174-b3d08cfb-e692-49d5-88ee-f9982d7bffab.png">
Not fully sure exactly the problem here, I haven't worked with django much before so I'm going to read into how manage.py works and hopefully understand more. 

Bug 2: Have hardly looked into this one, it may be related to bug 2 (at least I hope so).
<img width="836" alt="image" src="https://user-images.githubusercontent.com/90227881/178528019-0a640247-55b6-48fc-bfee-b339630c3088.png">
<img width="683" alt="image" src="https://user-images.githubusercontent.com/90227881/178528163-63f7bdb8-7436-4915-bdde-08f98101efd8.png">

Will focus on the first bug and go through these one by one.

**Comment by sabrina0822** [comment]
@malajvan was able to figure out that a big reason why jobs weren't loading was due to a change from the way __metaclass__ works in python 2 to python 3. In python 3 it needs to be a [parameter of the RodanTask](https://github.com/DDMAL/Rodan/commit/35d042f3554c40f8fc7d2a2ee98ee1a77412e29e) function as opposed to just being within the function. 

Other minor python 2 to 3 changes including the type of map/filter outputes (no longer a list in python 3) and the lack of a Type class (type.StringTypes --> ,str).

Jobs now load and are ready for testing

**Comment by fujinaga** [comment]
Bravo to both @sabrina0822 and @malajvan !

**Comment by timothydereuse** [comment]
Can this issue be closed, since things are building again?

**Comment by malajvan** [comment]
@timothydereuse this is still open because it's intended for all errors regarding rodan-main migration to python3. I suggest it's best to wait until we merge the PR to develop to close this issue

**Comment by timothydereuse** [comment]
Ah, okay. Was confused by the title. Will change it slightly for my own sake.

**Comment by sabrina0822** [comment]
Newest bug found while testing: 
PNG image resource types aren't properly identified, instead they recieve 'application/octet-stream' as their resource type.

<img width="601" alt="image" src="https://user-images.githubusercontent.com/90227881/180817589-eb1f2533-8b1d-418a-99f5-3f697d6bb31d.png">
This doesn't happen with any other resource type (at least none that we've seen so far)

I noticed when you run rodan-main for the first time and all resource types are being added this happens for the image/rgba+png type.
<img width="1124" alt="image" src="https://user-images.githubusercontent.com/90227881/180818072-c03d25e4-7207-49bf-b4cf-5546409859ea.png">

My best guess is that since the pixel_wrappe's input ports were changed recently that there are conflicting description for the rgba+png resource type. Not fully sure though since this issue doesn't happen on staging. 
Will consult with the document analysis team
